### PR TITLE
Fix/from system dimensionality

### PR DESCRIPTION
**Pull request by vyasr**
## Description
I adapted `from_system` to correctly handle inputs from various data sources when the systems are 2-dimensional. I also fixed an error in how Ovito inputs were handled.

## Motivation and Context

Currently the code simply passes the underlying object from various inputs through. This method doesn't work for 2D systems unless the object actually provides a `dimensions` attribute. Additionally, the current version will be a bit noisy for HOOMD outputs that set `Lz=1` for 2D systems, so we need to construct the box explicitly to silence these. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [x] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/credits.rst).
- [x] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ The format is based on
 and this project adheres to
 [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## next
+
+### Fixed
+* Conversion of automatic conversion of 2D systems from various data sources.
```
Comment (bdice) [inline ChangeLog.md]:
```suggestion
* Automatic conversion of 2D systems from various data sources.
```

File: freud/locality.pyx [hunk 2]
```diff
@@ -311,11 +315,21 @@ cdef class NeighborQuery:
 
         # MDAnalysis compatibility
         elif match_class_path(system, 'MDAnalysis.coordinates.base.Timestep'):
+            if dimensions is None:
+                logger.warn("Assuming MDAnalysis timestep is a 3-dimensional "
+                            "system. You must specify the dimensionality in "
+                            "from_system (or manually convert the frame's box "
```
Comment (bdice) [inline freud/locality.pyx]: Manually convert the frame's box? That doesn't make sense to me.

Also I asked Oliver Beckstein just now and MDAnalysis doesn't formally support 2D systems.

File: freud/locality.pyx [hunk 3]
```diff
@@ -324,12 +338,18 @@ cdef class NeighborQuery:
...
+                dimensions=2 if system.cell.is2D else 3)
+            system = (box, system.particles.positions)
 
         # HOOMD-blue snapshot compatibility
         elif (hasattr(system, 'box') and hasattr(system, 'particles') and
               hasattr(system.particles, 'position')):
-            system = (system.box, system.particles.position)
+            # Explicitly construct the box to silence warnings from box
+            # constructor because HOOMD sets Lz=1 rather than 0 for 2D boxes.
+            box = freud.Box(system.box.Lx, system.box.Ly, xy=system.box.xy)
```
Comment (bdice) [inline freud/locality.pyx]: Seems like this breaks 3D compatibility?

**Review by bdice — CHANGES_REQUESTED** [review]
Changes requested.

**Review by bdice — CHANGES_REQUESTED** [review]
Changes requested.

File: freud/locality.pyx (hunk 2 above)
Comment (vyasr) [reply freud/locality.pyx]: OK, then I'll just remove this warning and we'll assume inputs are 3D.

File: freud/locality.pyx (hunk 3 above)
Comment (vyasr) [reply freud/locality.pyx]: Good catch, yes.

**Review by bdice — APPROVED** [review]
LGTM!

**Pull request merged**